### PR TITLE
fix: allow setsize to be called within a move or resize for preventDefault

### LINKED ISSUE #34599: [Bug]: setBounds calls during will-resize are no longer applied when preventing the default resize
### Preflight Checklist

- [X] I have read the [Contributing Guidelines](https://github.com/electron/electron/blob/main/CONTRIBUTING.md) for this project.
- [X] I agree to follow the [Code of Conduct](https://github.com/electron/electron/blob/main/CODE_OF_CONDUCT.md) that this project adheres to.
- [X] I have searched the [issue tracker](https://www.github.com/electron/electron/issues) for a bug report that matches the one I want to file, without success.

### Electron Version

17.4, 18, 19

### What operating system are you using?

Windows

### Operating System Version

Windows 10 Pro 21H1

### What arch are you using?

x64

### Last Known Working Electron version

17.3.1

### Expected Behavior

Prior to version 17.4.0, it was possible to override the bounds received within will-resize event. While this ability is not explicitly documented, we rely on it for window management within our product.

### Actual Behavior

The window no longer resizes when the will-resize event is prevented and we attempt to control the resize with setBounds. 

### Testcase Gist URL

https://gist.github.com/meredith-ciq/a7ad120e84bd028a3fbd26b8f61aa45c

### Additional Information

This was caused during the changes from:
[#33288](https://github.com/electron/electron/pull/33288)
I believe the gate on is_resizing_ at the top of setBounds is preventing any ability to overwrite the bounds that come in during a will-resize event.

Our application currently overrides the native resize and does our own resize after modifying the window bounds. This has caused resize to break in recent electron versions as the intermediate setBounds calls during resize are not being applied. We have several aspects of window management that depend on having this level of control over resize:

- Groups of windows maintaining proportions when a single window in the group is resized
- Snapping edges of windows to other windows during resize
- Resolving jitter of a window during resize by applying our own calculations

 I've also mentioned this issue in a related issue: #33897. The fix from that issue only had the effect of not applying the final setBounds in the resize action. In our case that causes the window to no longer resize instead of giving no visual indication that a resize was occurring until the corner or edge of the window was dropped. I did some testing with the gist in that issue on the latest electron and it appears the behavior only works when some of the native resize events are allowed through.

### PR DESCRIPTION
**Pull request by mesner**
#### Description of Change

Closes #34599
Refs #33288
Refs #33279

The prior work in #33279 to allow calling SetBounds during a resize or move (defined as time period between receiving a WM_MOVING or WM_SIZING--respectively--and WM_EXITSIZEMOVE WNDPROC message) prevented the ability to call preventDefault() within 'will-resize' or 'will-move' and then call SetBounds to change the size and position of the window immediately.  After the PR, the window does not change bounds until the user ends the resize or move by releasing the mouse button, e.g. This PR simply removes the early-return statement to forward the SetBounds call to the widget, allowing for the bounds to change immediately if the movement or resizing has been cancelled through 'preventDefault' or after it has ended otherwise.

Tested with https://gist.github.com/6b6ea5cdaa1883656a8bd0bf89195283 and https://gist.github.com/meredith-ciq/a7ad120e84bd028a3fbd26b8f61aa45c

This is unfortunately not directly testable via spec, given that it's dependent on user-initiated movement.

#### Checklist

- [x] PR description included and stakeholders cc'd
- [x] [PR release notes](https://github.com/electron/clerk/blob/master/README.md) describe the change in a way relevant to app developers, and are [capitalized, punctuated, and past tense](https://github.com/electron/clerk/blob/master/README.md#examples).

#### Release Notes

Notes: Fixed an issue in which calling setBounds() after e.preventDefault in a 'will-move' or 'will-resize' event wouldn't change the window's shape until the mouse button was released.

### CODE REVIEW

**Comment by mesner** [comment]
@codebytere May I have some help identifying the next steps for this PR?

**Review by zcbenz — APPROVED** [review]
To conclude, https://github.com/electron/electron/pull/33288 assumes `widget()->SetBounds(bounds)` is always ignored by Chromium when window is moving or resizing, but https://github.com/electron/electron/issues/34599 says it is still working under some cases.

If I'm understanding it correctly then I'm good with this PR, apparently removing the early return won't affect the fix of #33288, and it will recover the old behavior for other cases.

**Comment by mesner** [comment]
@zcbenz Yes, you're assessment is correct. Thanks!

**Comment by mesner** [comment]
@zcbenz What is needed for this to be merged?  Sorry to pester but we are stuck on v17.0.0 until #34599 can be resolved.

**Comment by zcbenz** [comment]
I was waiting for another review, but this change is obvious enough and I'm going to just merge.

**Pull request merged**
